### PR TITLE
Add `Lint/NotNilAfterNoBang` rule

### LINKED ISSUE #283: Warning on `find/index(...).not_nil!`
Since they could be written as `find/index!(...)`

### PR DESCRIPTION
**Pull request by Sija**
Resolves #283

### CODE REVIEW

File: src/ameba/rule/lint/not_nil_with_no_bang.cr [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+      return unless name_location = obj.name_location
+      return unless name_location_end = name_end_location(obj)
+      return unless end_location = name_end_location(node)
+
+      msg = MSG % {obj.name, obj.name}
+
+      issue_for name_location, end_location, msg do |corrector|
+        corrector.insert_after(name_location_end, '!')
+        corrector.remove_trailing(node, ".not_nil!".size)
```
Comment (veelenga) [inline src/ameba/rule/lint/not_nil_with_no_bang.cr:57]: What about using `NOT_NIL_NAME` here?

```suggestion
        corrector.remove_trailing(node, NOT_NIL_NAME.size + 1)
```

**Review by veelenga — APPROVED** [review]
Looks nice and clean. I have just one non-blocking comment

File: src/ameba/rule/lint/not_nil_with_no_bang.cr (hunk 1 above)
Comment (Sija) [reply src/ameba/rule/lint/not_nil_with_no_bang.cr:57]: I had it done exactly like that before, I've changed it though to make it more obvious what is it about 😅

**Comment by Sija** [comment]
I'm not sure about the name though, any suggestions?

**Comment by veelenga** [comment]
@Sija maybe one of these?

```
ChainedNotNill
ChainedUnneededNotNill
ChainedNotNillAfterNoBang
```

I would probably pick the first one (the smallest though). But you decide

**Comment by Sija** [comment]
I've finally settled on `NotNilAfterNoBang`, thanks for the suggestions :)

**Pull request merged**